runner/app/live/streamer/process.py

```diff
@@ -172,57 +172,86 @@ async def _initialize_pipeline(self):
                     raise
 
     async def _run_pipeline_loops(self):
-        pipeline = await self._initialize_pipeline()
-        input_task = asyncio.create_task(self._input_loop(pipeline))
-        output_task = asyncio.create_task(self._output_loop(pipeline))
-        param_task = asyncio.create_task(self._param_update_loop(pipeline))
+        self.pipeline = await self._initialize_pipeline()
+        input_task = asyncio.create_task(self._input_loop())
+        output_task = asyncio.create_task(self._output_loop())
+        param_task = asyncio.create_task(self._param_update_loop())
 
         async def wait_for_stop():
             while not self.is_done():
                 await asyncio.sleep(0.1)
-
-        tasks = [input_task, output_task, param_task, asyncio.create_task(wait_for_stop())]
+            logging.info("PipelineProcess: Pipeline loops interrupted, stopping pipeline")
 
         try:
-            await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
+            self.tasks = [input_task, output_task, param_task, asyncio.create_task(wait_for_stop())]
+            
+            # Wait for first task to complete or fail
+            done, pending = await asyncio.wait(self.tasks, return_when=asyncio.FIRST_COMPLETED)
+            
+            logging.info("PipelineProcess: Pipeline shutting down") 
+            
+            # Check if any task failed and get the error details
+            results = await asyncio.gather(*done, return_exceptions=True)
+            for task, result in zip(done, results):
+                if isinstance(result, Exception):
+                    task_name = task.get_name() if task.get_name() else 'unknown'
+                    self._report_error(f"Error while cancelling {task_name} task: {str(result)}")
+                    self.done.set()
+            
+            # Cancel remaining tasks and gather results
+            if pending:
+                results = await asyncio.gather(*pending, return_exceptions=True)
+                for task, result in zip(pending, results):
+                    if isinstance(result, Exception):
+                        task_name = task.get_name() if task.get_name() else 'unknown'
+                        self._report_error(f"Error while cancelling {task_name} task: {str(result)}")
+                        self.done.set()
+                
         except Exception as e:
-            self._report_error(f"Error in pipeline loops: {e}")
+            self._report_error(f"Fatal error in pipeline loops: {str(e)}")
+            self.done.set()
+            raise  # Re-raise to ensure the error is propagated
```

Comment:
No one will catch this, this is the main function ran by the process, and we already have the log above (which should be critical if we ever see in production). I think it's better to remove the `raise` to avoid any confusion.